src/relay/qnn/op/quantize.cc

```diff
@@ -46,19 +47,22 @@ bool QuantizeRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   }
 
   const auto input_dtype = data->dtype;
-  ICHECK(input_dtype == DataType::Float(32))
-      << "Input type should be one of float32 but was " << input_dtype;
+  ICHECK(input_dtype == DataType::Float(32) || input_dtype == DataType::Int(32))
```

Comment:
I have no idea why the input to quantize could be int32.

Reply:
Sure but it is the output that needs to be int32. Here, you are allowing inputs to be int32. Am I missing something?